bundles/org.openhab.ui/web/src/components/widgets/widget-foreground-service.js

```diff
@@ -0,0 +1,47 @@
+export default {
+  data () {
+    return {
+      pageEl: null,
+      inForeground: false
+    }
+  },
+  mounted () {
+    // determine the current page - support: https://caniuse.com/?search=closest
+    this.pageEl = this.$el.closest('.page')
+    const isPopup = this.$el.closest('.popup') !== null
+    // start the foreground activity immediately if the page
+    // is already in the foreground when the widget is mounted
+    // or is being shown within a popup
+    if (this.pageEl.classList.contains('page-current') || isPopup) {
+      this.inForeground = true
+      this.startForegroundActivity()
+    }
+    this.$f7.on('pageAfterIn', this.onPageAfterIn)
+    this.$f7.on('pageBeforeOut', this.onPageBeforeOut)
```

Comment:
> Yeah, that was the first thing i tried, but it led to some package conflict which i thought could happen when your npm cache is out of sync (it was not the issue in this case), so when trying from a fresh clone failed, I kinda gave up debugging the issue :-)

This should probably be addressed asap as it's only useful to @stefan-hoehn at the moment and breaks the build for others.

> So does that mean this technique is not viable, or needs to be tweaked in your opinion?

It's just a matter of making it work in as many combinations of pages & modals as possible. It would appear at mounting time modals are in the DOM under a background div with the `framework7-modals` class (then they're moved elsewhere). There's no need to subscribe to events when in a modal, since they are temporary in nature and destroyed when closed. A caveat is that a modal may be covered by another modal (or navigate to a page while still in the background) but in this case the foreground activity won't stop.
The suggestion above seems to work in most cases:

![chrome_2022-06-10_19-46-00](https://user-images.githubusercontent.com/2004147/173122609-cc6ab5ff-d629-4eb8-8f4f-6044119eae79.gif)